.github/scripts/cherry-pick-helper.sh

```diff
@@ -0,0 +1,89 @@
+#!/bin/bash
+# .github/scripts/cherry-pick-helper.sh
+# Utility script for manual cherry-pick operations
+
+set -euo pipefail
+
+COMMIT_SHA="$1"
+TARGET_BRANCH="$2"
+SHORT_SHA="${COMMIT_SHA:0:7}"
+BRANCH_NAME="cherry-pick/${SHORT_SHA}-to-${TARGET_BRANCH}"
+
+# Colors for output
+RED='\033[0;31m'
+GREEN='\033[0;32m'
+YELLOW='\033[1;33m'
+NC='\033[0m' # No Color
+
+log_info() { echo -e "${GREEN}ℹ️  $1${NC}"; }
+log_warn() { echo -e "${YELLOW}⚠️  $1${NC}"; }
+log_error() { echo -e "${RED}❌ $1${NC}"; }
+
+# Validate inputs
+if [[ -z "$COMMIT_SHA" || -z "$TARGET_BRANCH" ]]; then
+    log_error "Usage: $0 <commit-sha> <target-branch>"
+    exit 1
+fi
+
+log_info "Cherry-picking ${SHORT_SHA} to ${TARGET_BRANCH}"
+
+# Update branches
+log_info "Fetching latest changes..."
+git fetch origin
+
+log_info "Checking out develop branch..."
+git checkout develop
+git pull origin develop
+
+log_info "Checking out target branch: ${TARGET_BRANCH}"
+if ! git checkout "${TARGET_BRANCH}"; then
+    log_error "Failed to checkout target branch: ${TARGET_BRANCH}"
+    exit 1
+fi
+git pull origin "${TARGET_BRANCH}"
+
+# Create cherry-pick branch
+log_info "Creating cherry-pick branch: ${BRANCH_NAME}"
+git checkout -b "${BRANCH_NAME}"
+
+# Get commit message for PR
+COMMIT_MSG=$(git log --format=%s -n 1 "${COMMIT_SHA}")
+
+# Attempt cherry-pick
+log_info "Attempting cherry-pick..."
+if git cherry-pick "${COMMIT_SHA}"; then
```

Comment:
```suggestion
if git cherry-pick -x "${COMMIT_SHA}"; then
```

`-x` will add the original commit sha in the commit message for the cherry picked commmit.

Reply:
We'll need to sign the cherry pick to make CI happy. Locally that's easy. From CI, we could use GitHub's REST API to sign commit but I'm not sure how that works with cherry-pick's. I can work with you on a solution here because I also need it for something else.